operate/webapp/src/test/java/io/camunda/operate/webapp/zeebe/operation/ResolveIncidentHandlerTest.java

```diff
@@ -0,0 +1,91 @@
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.camunda.client.CamundaClient;
+import io.camunda.client.impl.CamundaClientFutureImpl;
+import io.camunda.client.impl.command.JobUpdateRetriesCommandImpl;
+import io.camunda.client.impl.command.ResolveIncidentCommandImpl;
+import io.camunda.operate.webapp.reader.IncidentReader;
+import io.camunda.webapps.schema.entities.operate.ErrorType;
+import io.camunda.webapps.schema.entities.operate.IncidentEntity;
+import io.camunda.webapps.schema.entities.operation.OperationEntity;
+import io.camunda.webapps.schema.entities.operation.OperationType;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InOrder;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class ResolveIncidentHandlerTest {
+
+  @Mock IncidentReader incidentReader;
+  @Spy CamundaClient camundaClient;
+  @Mock JobUpdateRetriesCommandImpl updateRetriesCommand;
+  @Mock ResolveIncidentCommandImpl resolveIncidentCommand;
+  @InjectMocks @Spy private ResolveIncidentHandler resolveIncidentHandler;
+
+  @Test
+  public void shouldUpdateRetriesBeforeResolvingTaskListenerIncident() throws Exception {
+    // given
+    final Long incidentKey = 123L;
+    final Long jobKey = 1L;
+    when(incidentReader.getIncidentById(incidentKey))
+        .thenReturn(
+            new IncidentEntity()
+                .setErrorType(ErrorType.TASK_LISTENER_NO_RETRIES)
+                .setJobKey(jobKey)
+                .setKey(incidentKey));
+
+    final String operationId = "456";
+    final OperationEntity operationEntity =
+        new OperationEntity()
+            .setType(OperationType.RESOLVE_INCIDENT)
+            .setIncidentKey(incidentKey)
+            .setId(operationId);
+
+    final CamundaClientFutureImpl future1 = new CamundaClientFutureImpl<>();
+    final CamundaClientFutureImpl future2 = new CamundaClientFutureImpl<>();
+    when(updateRetriesCommand.send()).thenReturn(future1);
+    when(resolveIncidentCommand.send()).thenReturn(future2);
+    future1.complete(null);
+    future2.complete(null);
+
+    when(updateRetriesCommand.operationReference(Long.parseLong(operationId)))
+        .thenReturn(updateRetriesCommand);
+    when(resolveIncidentCommand.operationReference(Long.parseLong(operationId)))
+        .thenReturn(resolveIncidentCommand);
+    when(updateRetriesCommand.retries(1)).thenReturn(updateRetriesCommand);
+
+    when(camundaClient.newUpdateRetriesCommand(jobKey)).thenReturn(updateRetriesCommand);
+    when(camundaClient.newResolveIncidentCommand(incidentKey)).thenReturn(resolveIncidentCommand);
+
+    doNothing().when(resolveIncidentHandler).markAsSent(any());
```

Comment:
I think this interferes too much with the implementation. Let's not modify the behavior of the class under test. Instead, rather mock the metrics and batch operation writer.